specs/commandline/sign.md

```diff
@@ -29,16 +29,18 @@ Usage:
   notation sign [flags] <reference>
 
 Flags:
-  -e,  --expiry duration          optional expiry that provides a "best by use" time for the artifact. The duration is specified in minutes(m) and/or hours(h). For example: 12h, 30m, 3h20m
-  -h,  --help                     help for sign
-       --image-spec string        manifest type for signatures. options: v1.1-artifact, v1.1-image (default: v1.1-artifact)
-  -k,  --key string               signing key name, for a key previously added to notation's key list.
-  -p,  --password string          password for registry operations (default to $NOTATION_PASSWORD if not specified)
-       --plain-http               registry access via plain HTTP
-       --plugin-config strings    {key}={value} pairs that are passed as it is to a plugin, refer plugin's documentation to set appropriate values
-       --signature-format string  signature envelope format, options: 'jws', 'cose' (default "jws")
-  -u,  --username string          username for registry operations (default to $NOTATION_USERNAME if not specified)
-  -m,  --user-metadata strings    {key}={value} pairs that are added to the signature payload
+  -d,  --debug                       debug mode
+  -e,  --expiry duration             optional expiry that provides a "best by use" time for the artifact. The duration is specified in minutes(m) and/or hours(h). For example: 12h, 30m, 3h20m
+  -h,  --help                        help for sign
+       --artifact-type string        manifest type for signatures. options: v1.1-artifact, v1.1-image (default "v1.1-artifact")
```

Comment:
According to the spec, should it be `--image-spec`?

Reply:
@priteshbandi I recall that you did agree with the name `--image-spec` during spec review, see https://github.com/notaryproject/notation/pull/502#discussion_r1069095292.
It is more flexible and extensible by this design `--image-spec <spec version>-<manifest type>`  that users can specify the spec version and the manifest type that they want to use. There could be new types in the future. On the other hand, the name of `--artifact-type` could cause confusion, since there is a field named `artifactType` defined in `OCI artifact manifest` spec. User may think this flag `--artifact-type` is to be used to configure the filed `artifactType`.
I suggest we keep the current definition in the spec.

Reply:
I changed back to `--image-spec` for now based on comments above.

Reply:
FWIW, I agree with @priteshbandi on this. I think users specifying both an image spec version and artifact type feels like too much implementation detail exposure.

Reply:
@priteshbandi @jlbutler Another option here is making it a boolean flag, for example, `--legacy-artifact-type`. Omitting the flag means using OCI artifact manifest to store signatures. Setting the flag means using OCI image manifest. What do you think?

Reply:
I vote for `--image-spec <spec version>-<manifest type>`. By specifying the OCI image spec version, it will be much clear as the artifact/image manifest may differentiate by spec versions.

Just FYI, ORAS has a similar situation with specifying a tag to custom the pushing behavior of the manifest type, see https://github.com/oras-project/oras/blob/main/docs/proposals/compatibility-mode.md#force-uploading-a-specific-manifest-type-using-a-flag---image-spec

Reply:
I think both `--image-spec` and `--artifact-type` as flag are confusing. I am struggling to understand the meaning of both. Also, from the help text above the expectation seems to be that this parameter will expect some value like `v1.0` or `v1.1`. This is too much details for somebody who doesn't even know what is the difference between OCI 1.0 and OCI 1.1 or has no knowledge of how different registries store the signature. 

The flow is the following:
- User tries to store the signature into the registry and gets an error.
- Based on the descriptiveness of the error, the user may want to retry the operation with the switch

There are two parts here:
- First, we need to make sure that the error the user sees is descriptive enough so they know what is happening. I would expect to get some error like: **Saving the signature failed. It is possible the registry doesn't support the format the signature is stored. Try --THE_SWITCH to use the legacy format for storing the signature**
- The name of the switch should be more user friendly and not requiring the user to know anything about the OCI spec. What comes to my mind is something like `--use-legacy-format`

Reply:
@toddysm Thanks. It makes sense. There is an existing flag named `--signature-format` for `JWS` or `COSE` envelope. So, it's better not to use the word `format` to avoid confusion. How about `--use-legacy-manifest`. Here is the summary for current proposals:
1. flag name `--signature-manifest`, values could be `1.0` or `1.1`
2. flag name `--signature-manifest`, values could be `image` or `artifact`
3. flag name `--use-legacy-format`, this is a Boolean type of flag
4. flag name `--use-legacy-manifest`, this is a Boolean type of flag

Reply:
One more thing, is it appropriate to use the wording `legacy` before OCI spec cut a 1.1 release?

Reply:
I see. So, what about `--use-image-manifest` Boolean flag? This is quite explicit, and it doesn't require deep knowledge of the OCI spec versioning.

Reply:
General agreement on the Notary community call on Feb 6 was to proceed with option 2 above using string values of `image` and `artifact`, `artifact` will be the default. cc @patrickzheng200